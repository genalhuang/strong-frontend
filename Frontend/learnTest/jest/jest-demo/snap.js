export const generateConfig = () => {
  return {
    server: 'http://localhost',
    port: 8082,
    // name: 'genal'
    time: new Date()
  }
}  