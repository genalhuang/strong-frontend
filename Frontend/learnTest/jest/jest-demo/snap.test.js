import { generateConfig } from "./snap";

// 静态的数值测试
// test('测试generateConfig函数',() => {
//   expect(generateConfig()).toEqual( {
//     server: 'http://localhost',
//     port: 8080,
//   })
// })

// 动态的数值测试
test("测试generateConfig函数", () => {
  expect(generateConfig()).toMatchSnapshot({
    time: expect.any(Date)
  });
});

// 行内快照 需要安装prettier
test("测试generateConfig函数", () => {
  expect(generateConfig()).toMatchInlineSnapshot(
    {
      time: expect.any(Date)
    },
    `
    Object {
      "port": 8082,
      "server": "http://localhost",
      "time": Any<Date>,
    }
  ` 
  );
});
