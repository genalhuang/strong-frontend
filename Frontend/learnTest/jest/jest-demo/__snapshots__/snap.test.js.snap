// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试generateConfig函数 1`] = `
Object {
  "port": 8082,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;
